src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -370,19 +370,25 @@ public function makePathRelative($endPath, $startPath)
         if ('\\' === DIRECTORY_SEPARATOR) {
             $endPath = str_replace('\\', '/', $endPath);
             $startPath = str_replace('\\', '/', $startPath);
-        }
 
-        $stripDriveLetter = function ($path) {
-            if (strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0])) {
-                return substr($path, 2);
-            }
+            $stripDriveLetter = function ($path) {
+                if (strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0])) {
+                    return array(substr($path, 2), $path[0].$path[1]);
+                }
 
-            return $path;
-        };
+                return $path;
+            };
 
-        $endPath = $stripDriveLetter($endPath);
-        $startPath = $stripDriveLetter($startPath);
+            $endPath = $stripDriveLetter($endPath);
+            $startPath = $stripDriveLetter($startPath);
 
+            if ($endPath[1] !== $startPath[1]) {
```

Comment:
This will do unexpected things if paths don't contain `:`. Array is returned only when it does. So when it doesn't, you are getting here second string character of relative path.